### DISCUSSION TITLE
Find/Grep within 1 or more selected directories

### DISCUSSION THREAD

**Discussion by alankritjoshi** [ibhagwan/fzf-lua#1406] [Q&A] [answered]
I use lazyvim and I confess I haven't learnt the advanced commands yet.

I want to achieve something something like a 2 step approach:

### Files

\<key for files\>

1. fzf window to search 1 or more directories <-- maybe triggered by fzf action (similar to https://github.com/ibhagwan/fzf-lua/discussions/1402)
2. fzf window to files within the selected directories

### Grep

\<key for grep\>

1. fzf window to search 1 or more directories <-- maybe triggered by fzf action
2. fzf window to grep within the selected directories

**Comment by ibhagwan** [comment, marked as answer]
Take a look at the commit notes for https://github.com/ibhagwan/fzf-lua/commit/98c1beb226783603d3759c5b39aa5d96e8e4d611.

You want to use `search_paths` for that.

**Reply by alankritjoshi** [reply]
This is amazing, thanks for making this!

**Reply by alankritjoshi** [reply]
```
    config = function()
      local fzf_lua = function()
        return require("fzf-lua")
      end

      fzf_lua().setup({
        files = {
          actions = {
            ["ctrl-e"] = {
              function(selected, opts)
                local files = {}
                for _, file in ipairs(selected) do
                  local entry = fzf_lua().path.entry_to_file(file)
                  table.insert(files, entry.stripped)
                end

                if #files == 0 then -- reset
                  fzf_lua().files({
                    fd_opts = opts.fd_opts,
                    prompt = opts.prompt,
                  })
                else
                  fzf_lua().live_grep({
                    search_paths = files,
                    prompt = "*Rg(" .. table.concat(files, ", ") .. ") >",
                  })
                end
              end,
            },
```

**Reply by ibhagwan** [reply]
Looks great @alankritjoshi!
